console/src/modules/chatroom/ChatroomListItem.tsx

```diff
@@ -45,17 +47,22 @@ export const ChatroomListItem: React.FC<ChatroomListItemProps> = ({
             callerPhoneNumber={chatroom.callerPhoneNumber}
           />
         </Box>
-        <IconButton onClick={() => setShowDetails(!showDetails)}>
-          {showDetails ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
-        </IconButton>
+
+        <Box
+          display="flex"
+          alignItems="flex-start"
+          justifyContent="space-between"
+        >
+          <IconButton onClick={() => setShowDetails(!showDetails)}>
+            {showDetails ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
+
+          <ResolveToggle chatroomId={chatroom.id} isResolved={chatroom.resolved} />
+        </Box>
       </Box>
+
       <Collapse in={showDetails}>
-        <Card sx={{ padding: 2 }}>
-          <Typography variant="body1">Description</Typography>
-          <Typography variant="body2">
-            {chatroom.description ?? "No description provided."}
-          </Typography>
-        </Card>
+        {showDetails && <ChatroomDetails chatroom={chatroom} />}
```

Comment:
`ChatroomDetails` renders `NotesSection`, which is where we mount the query to fetch the list of notes for a specific chatroom. So I added the `showDetails &&` conditional here to prevent fetching notes for all chatrooms when the chatroom cards are initially rendered, because that's a lot of work fetching data upfront that the user might not even look at. Instead, we'll only fetch the notes for a chatroom once that chatroom is expanded in the UI.

As a result of this change, we do miss out on the animation of collapsing the chatroom card, but I personally didn't think that animation was super important, and almost prefer the card collapsing in a snappier way. I could have passed down `showDetails` to `ChatroomDetails` to specifically not render just the `NotesSection` part, but that actually makes the animation look a bit more choppy.